kasa/modules/antitheft.py

```diff
@@ -7,3 +7,15 @@ class Antitheft(RuleModule):
 
     This shares the functionality among other rule-based modules.
     """
+
+    @property
+    def is_supported(self) -> bool:
+        """Return whether the module is supported by the device."""
+        if (
+            (is_supported := super().is_supported)
+            and (module_data := self._device._last_update.get(self._module))
+            and (get_next_action := module_data.get("get_next_action"))
+            and "err_code" in get_next_action
```

Comment:
Instead of using a nested if, I think it'd be nicer to have it separate. Just in case someone comes looking to implement special handling for lightstrips, c.f., my comment below.

Reply:
I refactored it in 7809f2809d9e7c842becfcc6353f1221ca26fb44
but I'm not 100% sure thats what you were getting at.

I didn't understand the use case for lightstrips

Reply:
Ah, so basically I was thinking that this feature is actually supported by the device but that we are calling an incorrect method, causing the "method not support" error. But it looks like the lightstrips may not support this "away mode" at all.

I'm getting that very same error on my KL130, but as `get_rules` is working I guessed that the `next_action` would return something only if the mode is active:
```
{'smartlife.iot.common.anti_theft': {'get_next_action': {'err_code': -2000,                                                           protocol.py:109
                                                                  'err_msg': 'Method '                                                                                           
                                                                             'not '                                                                                              
                                                                             'support'},                                                                                         
                                              'get_rules': {'enable': 0,                                                                                                         
                                                            'err_code': 0,                                                                                                       
                                                            'rule_list': []}},
```

Do you see response for the `get_rules` method on that module? `kasa --host asdf raw-command smartlife.iot.common.anti_theft get_rules`

Reply:
```
bdraco@MacBook-Pro-9 python-kasa % kasa --host 192.168.107.148 raw-command smartlife.iot.common.anti_theft get_rules
No --type defined, discovering..
Got error: SmartDeviceException("Error on smartlife.iot.common.anti_theft.get_rules: {'err_code': -1, 'err_msg': 'module not support'}")

```
```
== TP-LINK_Power Strip_542B - HS300(US) ==
        Host: 192.168.107.148
        Port: 9999
        Device state: True
        == Plugs ==
        * Socket 'Kasa_Smart Plug_542B_0' state: True since 2023-10-07 16:00:11
        * Socket 'Kasa_Smart Plug_542B_1' state: True since 2023-10-07 13:35:00
        * Socket 'Kasa_Smart Plug_542B_2' state: True since 2023-10-07 13:34:23
        * Socket 'Kasa_Smart Plug_542B_3' state: True since 2023-10-07 13:34:23
        * Socket 'Kasa_Smart Plug_542B_4' state: True since 2023-10-07 13:34:23
        * Socket 'Kasa_Smart Plug_542B_5' state: True since 2023-10-07 13:34:23

        == Generic information ==
        Time:         2023-10-07 18:40:39 (tz: {'index': 13, 'err_code': 0}
        Hardware:     2.0
        Software:     1.0.3 Build 201203 Rel.165457
        MAC (rssi):   C0:06:C3:42:54:2B (-40)
        Location:     {'latitude': 29.7852, 'longitude': -95.4073}

        == Device specific information ==
        LED state: True
        Childs count: 6
        On since: 2023-10-07 16:00:11

        == Current State ==
        <EmeterStatus power=1.015 voltage=118.499 current=0.021 total=0.004>

        == Modules ==
        + <Module Antitheft (anti_theft) for 192.168.107.148>
        + <Module Schedule (schedule) for 192.168.107.148>
        + <Module Usage (schedule) for 192.168.107.148>
        + <Module Time (time) for 192.168.107.148>
        - <Module Countdown (countdown) for 192.168.107.148>
        + <Module Emeter (emeter) for 192.168.107.148>
```

Reply:
For KP303 (uk three-socket strip) it's called just `anti_theft`, the `smartlife.iot.common`-prefixed are something for bulbs & leds.
```
❯ kasa --host 192.168.xx.xx raw-command anti_theft get_rules
No --type defined, discovering..
{"rule_list": [], "version": 2, "enable": 0}
```
```
❯ kasa --json --host 192.168.xx.xx state |jq .system.get_sysinfo.model
"KP303(UK)"
```

Reply:
```
kasa --host 192.168.107.148 raw-command antitheft get_rules
No --type defined, discovering..
Got error: SmartDeviceException("Error on antitheft.get_rules: {'err_code': -1, 'err_msg': 'module not support'}")
``
```
 kasa --json --host 192.168.107.148 state |jq .system.get_sysinfo.model
"HS300(US)"

```